portmaster/install/linux.md

```diff
@@ -153,6 +156,30 @@ sudo systemctl enable --now portmaster
 
 __6.__ Enjoy!
 
+### Security-Enhanced Linux (SELinux)
+
+If you are running with `SELINUX=enforcing` you probably wasn't successful with running portmaster, and you see following error in your `journalctl -u portmaster`:
+
+```
+dub 16 22:09:10 dev-fedora systemd[1]: Started Portmaster Privacy App.
+dub 16 22:09:10 dev-fedora systemd[30591]: portmaster.service: Failed to execute command: Permission denied
+dub 16 22:09:10 dev-fedora systemd[30591]: portmaster.service: Failed at step EXEC spawning /var/lib/portmaster/portmaster-start: Permission denied
+dub 16 22:09:10 dev-fedora systemd[1]: portmaster.service: Main process exited, code=exited, status=203/EXEC
+```
+
+This happened because SELinux won't let you to run binary from `/var/lib/portmaster` as systemd service. To this you need to change SELinux security context type of `portmaster-start` binary using following command:
+
+```bash
+sudo chcon -t bin_t /var/lib/portmaster/portmaster-start
```

Comment:
I think this is a great solution!

However, I am not sure if this wouldn't break if the Portmaster updates `portmaster-start` to a new version. Is the security context bound to the file and thus removed when the file is replaced with a new version?

Maybe, @ppacher has an idea how this works and how to handle this.

This could also be easily tested by `mv`ing a fresh copy of `portmater-start` to `/var/lib/portmaster/portmaster-start`. I think this comes closest to how the Portmaster would update the file.

Reply:
@dhaavi nice catch! can you please point me to url of an older `portmaster-start` version so I can use it to test it? when I cp the current `portmaster-start` it seems it brings SELinux context to `portmaster-start` copy.

Maybe I could use `rsync` and disable SELinux attrs copying in it but I haven't checked whether this is possible with rsync.

Reply:
Hey, I didn't see your comment, sorry.

Here you can get v0.5.1 of the portmaster-start binary: https://updates.safing.io/linux_amd64/start/portmaster-start_v0-5-1

Reply:
@northys  Any update on this? Does it work with that old version?

Reply:
@dhaavi it works I think

```

Reply:
@dhaavi it looks like both `portmaster-start` and `portmaster-start.old` binaries are now reporting the same build date so I think it updated itself on restart above

```
[root@northys-fedora portmaster]# ./portmaster-start version
Portmaster Start
version 0.5.4

commit tags/v0.6.14-0-g82a7350f9afee91f9948912857bbf943969f9b81
built with go1.15.8 (gc) linux/amd64
  using options 
  by user@docker
  on 18.05.2021

Licensed under the AGPLv3 license.
The source code is available here: https://github.com/safing/portmaster
[root@northys-fedora portmaster]# ./portmaster-start.old version
Portmaster Start
version 0.5.4

commit tags/v0.6.14-0-g82a7350f9afee91f9948912857bbf943969f9b81
built with go1.15.8 (gc) linux/amd64
  using options 
  by user@docker
  on 18.05.2021

Licensed under the AGPLv3 license.
The source code is available here: https://github.com/safing/portmaster
```

Reply:
@dhaavi yes I can confirm it updated itself. see log:

```
[root@northys-fedora portmaster]# journalctl -u portmaster.service --since "5 minutes ago" | grep -i updates:
kvě 29 23:00:11 northys-fedora portmaster-start[44973]: 210529 23:00:11.078 r/updating:096 ▶ INFO 088 updates: updated index stable.json
kvě 29 23:00:11 northys-fedora portmaster-start[44973]: 210529 23:00:11.102 r/updating:096 ▶ INFO 089 updates: updated index all/intel/intel.json
kvě 29 23:00:11 northys-fedora portmaster-start[44973]: 210529 23:00:11.102 r/updating:127 ▶ INFO 090 updates: everything up to date
kvě 29 23:00:11 northys-fedora portmaster-start[44973]: 210529 23:00:11.182 s/upgrader:349 ▶ INFO 091 updates: upgraded /opt/portmaster/portmaster-start to v0.5.4
kvě 29 23:00:11 northys-fedora portmaster-start[44973]: 210529 23:00:11.182 r/electron:052 ▶ INFO 092 updates: fixed SUID permission for chrome-sandbox
```

Reply:
@shaurya-007 done

Reply:
Yay!!!!!! When will the download be available, any guesses???

Reply:
@shaurya-007 I think you misunderstood the PR. This PR is about documentation about how to run portmaster in fedora but you have to download and install it manually. RPM is discussed in the other issue (safing/portmaster-packaging#4). However you can download and run portmaster manually. It's just about creating one directory, setting SELinux mode on portmaster-start and creating one systemd unit file. See more in the doc in this PR.

Reply:
@shaurya-007 btw you don't have to worry about updates! portmaster updates itself on each startup automatically so basically you don't need RPM to have portmaster on your fedora =)

Reply:
Great thanks for checking this @northys! I think we can then move forward with merging this! 🎉